src/UserService.Business/Commands/Certificate/RemoveCertificateCommand.cs

```diff
@@ -1,56 +1,111 @@
+    private readonly ILogger<RemoveCertificateCommand> _logger;
+
+    public RemoveCertificateCommand(
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IUserRepository userRepository,
+      ICertificateRepository certificateRepository,
+      IRequestClient<IRemoveImagesRequest> removeImagesRequest,
+      ILogger<RemoveCertificateCommand> logger)
     {
-        private readonly IAccessValidator _accessValidator;
-        private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IUserRepository _userRepository;
-        private readonly ICertificateRepository _certificateRepository;
-
-        public RemoveCertificateCommand(
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IUserRepository userRepository,
-            ICertificateRepository certificateRepository)
-        {
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _userRepository = userRepository;
-            _certificateRepository = certificateRepository;
-        }
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _certificateRepository = certificateRepository;
+      _removeImagesRequest = removeImagesRequest;
+      _logger = logger;
+    }
+
+    public async Task<OperationResultResponse<bool>> Execute(Guid certificateId)
+    {
+      OperationResultResponse<bool> result = new();
+      List<string> resultErrors = result.Errors;
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser sender = _userRepository.Get(senderId);
+      DbUserCertificate userCertificate = _certificateRepository.Get(certificateId);
+
+      if (!(sender.IsAdmin ||
+            _accessValidator.HasRights(Rights.AddEditRemoveUsers))
+            && senderId != userCertificate.UserId)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+
+        resultErrors.Add("Not enough rights");
+        result.Status = OperationResultStatusType.Failed;
+
+        return result;
+      }
 
-        public OperationResultResponse<bool> Execute(Guid certificateId)
+      ICollection<DbUserCertificateImage> imagesToRemove = userCertificate.Images;
+
+      if (imagesToRemove.Any())
+      {
+        await RemoveImages(imagesToRemove, resultErrors);
+      }
+      
+      result.Body = _certificateRepository.Remove(userCertificate);
+      result.Status = resultErrors.Any() ? OperationResultStatusType.PartialSuccess : OperationResultStatusType.FullSuccess;
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+
+      return result;
+    }
+
+    private async Task<bool> RemoveImages(ICollection<DbUserCertificateImage> imagesToRemove, ICollection<string> errors)
```

Comment:
we use style when private methods are before publics

Reply:
its bad(